compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2107,8 +2107,11 @@ class TrackingTypeComparer(initctx: Context) extends TypeComparer(initctx) {
   }
 
   override def gadtBounds(sym: Symbol)(implicit ctx: Context): TypeBounds = {
-    footprint += sym.typeRef
-    super.gadtBounds(sym)
+    if (!sym.exists) null
+    else {
+      footprint += sym.typeRef
+      super.gadtBounds(sym)
+    }
```

Comment:
IIUC that the problem is with `NoSymbol.typeRef` exploding, I'd prefer
```scala
if (sym.exists) footprint += sym.typeRef
super.gadtBounds(sym)
```
instead, because 1) this matters for logging 2) it should ultimately be the business of `GadtConstraint` to decide what to return for unconstrained symbols.

Otherwise LGTM.

Reply:
That looks fine.

There's similar code in the neighbouring `gadtAddLowerBound` and `gadtAddUpperBound` ... make the same change there as well?

Reply:
I've gone ahead and done that. Will merge when green.